src/framework/components/joint/component.js

```diff
@@ -0,0 +1,371 @@
+            constraint.setLinearLowerLimit(limits);
+            limits.setValue(llx.y, lly.y, llz.y);
+            constraint.setLinearUpperLimit(limits);
+            Ammo.destroy(limits);
+        }
+    }
+
+    _createConstraint() {
+        if (this._entityA && this._entityA.rigidbody) {
+            this._destroyConstraint();
+
+            const mat = new Mat4();
+
+            const bodyA = this._entityA.rigidbody.body;
+            bodyA.activate();
+
+            const jointWtm = this.entity.getWorldTransform();
+
+            const entityAWtm = this._entityA.getWorldTransform();
+            const invEntityAWtm = entityAWtm.clone().invert();
+            mat.mul2(invEntityAWtm, jointWtm);
+
+            const frameA = new Ammo.btTransform();
+            this._convertTransform(mat, frameA);
+
+            if (this._entityB && this._entityB.rigidbody) {
+                const bodyB = this._entityB.rigidbody.body;
+                bodyB.activate();
+
+                const entityBWtm = this._entityB.getWorldTransform();
+                const invEntityBWtm = entityBWtm.clone().invert();
+                mat.mul2(invEntityBWtm, jointWtm);
+
+                const frameB = new Ammo.btTransform();
+                this._convertTransform(mat, frameB);
+
+                this._constraint = new Ammo.btGeneric6DofSpringConstraint(bodyA, bodyB, frameA, frameB, !this._enableCollision);
+
+                Ammo.destroy(frameB);
+            } else {
+                this._constraint = new Ammo.btGeneric6DofSpringConstraint(bodyA, frameA, !this._enableCollision);
+            }
+
+            Ammo.destroy(frameA);
+
+            const axis = ['X', 'Y', 'Z', 'X', 'Y', 'Z'];
+
+            for (let i = 0; i < 6; i++) {
+                const type = i < 3 ? 'linear' : 'angular';
+                this._constraint.enableSpring(i, this['_' + type + 'Spring' + axis[i]]);
+                this._constraint.setDamping(i, this['_' + type + 'Damping' + axis[i]]);
+                this._constraint.setEquilibriumPoint(i, this['_' + type + 'Equilibrium' + axis[i]]);
+                this._constraint.setStiffness(i, this['_' + type + 'Stiffness' + axis[i]]);
+            }
+
+            this._constraint.setBreakingImpulseThreshold(this._breakForce);
+
+            this._updateLinearLimits();
+            this._updateAngularLimits();
+
+            const app = this.system.app;
+            const dynamicsWorld = app.systems.rigidbody.dynamicsWorld;
+            dynamicsWorld.addConstraint(this._constraint, !this._enableCollision);
+        }
+    }
+
+    _destroyConstraint() {
+        if (this._constraint) {
+            const app = this.system.app;
+            const dynamicsWorld = app.systems.rigidbody.dynamicsWorld;
+            dynamicsWorld.removeConstraint(this._constraint);
+
+            Ammo.destroy(this._constraint);
+            this._constraint = null;
+        }
+    }
+
+    initFromData(data) {
+        const props = [
```

Comment:
could make static to avoid getting initialized multiple times

Reply:
Done.